src/libraries/System.IO.FileSystem/tests/Directory/SymbolicLinks.cs

```diff
@@ -1,15 +1,59 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-using System.Collections.Generic;
 using System.Linq;
+using Microsoft.DotNet.RemoteExecutor;
 using Xunit;
 
 namespace System.IO.Tests
 {
-    public class Directory_SymbolicLinks : BaseSymbolicLinks
+    public class Directory_SymbolicLinks : BaseSymbolicLinks_FileSystem
     {
-        [ConditionalFact(nameof(CanCreateSymbolicLinks))]
+        protected override void CreateFileOrDirectory(string path, bool createOpposite = false)
+        {
+            if (!createOpposite)
+            {
+                Directory.CreateDirectory(path);
+            }
+            else
+            {
+                File.Create(path).Dispose();
+            }
+        }
+
+        protected override FileSystemInfo CreateSymbolicLink(string path, string pathToTarget) =>
+            Directory.CreateSymbolicLink(path, pathToTarget);
+
+        protected override FileSystemInfo ResolveLinkTarget(string linkPath, bool returnFinalTarget = false) =>
+            Directory.ResolveLinkTarget(linkPath, returnFinalTarget);
+
+        protected override void AssertIsCorrectTypeAndDirectoryAttribute(FileSystemInfo linkInfo)
+        {
+            if (linkInfo.Exists)
+            {
+                Assert.True(linkInfo.Attributes.HasFlag(FileAttributes.Directory));
+            }
+            Assert.True(linkInfo is DirectoryInfo);
+        }
+
+        protected override void AssertLinkExists(FileSystemInfo link)
+        {
+            if (PlatformDetection.IsWindows)
+            {
+                Assert.True(link.Exists);
+            }
+            else
+            {
+                // Unix implementation detail:
+                // When the directory target does not exist FileStatus.GetExists returns false because:
+                // - We check _exists (which whould be true because the link itself exists).
+                // - We check InitiallyDirectory, which is the initial expected object type (which would be true).
+                // - We check _directory (false because the target directory does not exist)
```

Comment:
Do I understand correctly that if we have a link to non-existing directory, `FileSystemInfo.Exists` returns true on Windows but false on Unix?

why the implementations are not aligned? This is confusing

Reply:
https://github.com/dotnet/runtime/blob/87cd70c4ea9f0452f52db7dc9fd5bb3564a2e9e6/src/libraries/System.Private.CoreLib/src/System/IO/FileStatus.Unix.cs#L235-L238
That's becuase the second expression evaluates as false.
* `InitiallyDirectory` is evaluated as `this is DirectoryInfo`.
* `_isDirectory` stays as `false` when the symbolic link is broken. 

The behavior is like that prior to our changes and if we try to change `new DirectoryInfo("path-to-broken-link").Exists` to return true if the link itself exists (regardless of being broken) it would be a breaking change.